common/lib/xmodule/xmodule/modulestore/tests/test_mixed_modulestore.py

```diff
@@ -523,6 +523,75 @@ def test_has_changes(self, default_ms):
         component = self.store.publish(component.location, self.user_id)
         self.assertFalse(self.store.has_changes(component))
 
+    @ddt.data('split')
+    def test_unit_stuck_in_draft_mode(self, default_ms):
+        """
+        After revert_to_published() the has_changes() should return false if draft has no changes
+        """
+        self.initdb(default_ms)
+
+        test_course = self.store.create_course('testx', 'GreekHero', 'test_run', self.user_id)
```

Comment:
@cpennington , mostly because the other tests are written in this pattern. I agree that CourseFactory is the clearer, cleaner way to write this test -- what's your thought about breaking the pattern of the tests around it though?